core/tests/protractor_utils/PreferencesPage.js

```diff
@@ -41,6 +42,44 @@ var PreferencesPage = function() {
     by.css('.protractor-test-creator-dashboard-radio'));
   var learnerDashboardRadio = element(
     by.css('.protractor-test-learner-dashboard-radio'));
+  var profilePhotoClickable = element(
+    by.css('.protractor-test-photo-clickable'));
+  var profilePhotoUploadInput = element(
+    by.css('.protractor-test-photo-upload-input'));
+  var profilePhotoSubmitButton = element(
+    by.css('.protractor-test-photo-upload-submit'));
+  var customProfilePhoto = element(
+    by.css('.protractor-test-custom-photo'));
+  var profilePhotoCropper = element(
+    by.css('.protractor-test-photo-crop'));
+  var profilePhotoUploadError = element(
+    by.css('.protractor-test-upload-error'));
+
+  this.expectUploadError = function() {
+    expect(profilePhotoUploadError.isDisplayed()).toBe(true);
+  };
+
+  this.uploadProfilePhoto = function(imgPath) {
+    profilePhotoClickable.click();
+    absPath = path.resolve(__dirname, imgPath);
+    return profilePhotoUploadInput.sendKeys(absPath);
+  };
+
+  this.submitProfilePhoto = function(imgPath) {
+    return this.uploadProfilePhoto(imgPath).then(function() {
+      waitFor.visibilityOf(
+        profilePhotoCropper, 'photo cropper still invisible');
+    }).then(function() {
+      profilePhotoSubmitButton.click();
+    }).then(function() {
+      return waitFor.invisibilityOf(
+        profilePhotoUploadInput, 'photo upload still visible');
```

Comment:
Same here! `Photo upload is taking too long to disappear`

Reply:
Done in 266e70f